pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -816,7 +816,14 @@ stdenv.mkDerivation ({
 # Works around https://github.com/NixOS/nixpkgs/issues/166205.
 # This can be dropped once a fix has been committed to cc-wrapper.
 // lib.optionalAttrs (stdenv.cc.isClang && stdenv.cc.libcxx != null) {
-  env.NIX_LDFLAGS = "-l${stdenv.cc.libcxx.cxxabi.libName}";
+  env = (
+    let
+      args_env = args.env or { };
```

Comment:
args never has the env attribute at the moment so this makes little sense.

`NIX_LDFLAGS` should be able to be set as a direct argument to `stdenv.mkDerivation` and be propagated to the environment which would avoid this whole dance.

Reply:
I’ll revise and push and update when I have some bandwidth later today.

Reply:
I guess we can just leave it as is on `staging-next`, the change I'd like would mean a mass rebuild I think.

Reply:
mass rebuilds are OK now.  We have no binaries there (again).

Reply:
I’d wanted to rebuild Cachix to confirm rebuilds, but it’s taking a while to get there, so I fixed and force-pushed. I left it at the end since that keeps the diff small and lets it remain separated for the clang/libc++ case.

Reply:
From what I can tell, moving it up to the top level doesn’t appear to have caused rebuilds locally.